crates/pathfinder/src/sequencer/reply.rs

```diff
@@ -260,17 +265,31 @@ pub mod transaction {
         #[serde_as(as = "Option<FeeAsHexStr>")]
         #[serde(default)]
         pub max_fee: Option<Fee>,
-        #[serde_as(as = "Option<Vec<TransactionSignatureElemAsDecimalStr>>")]
         #[serde(default)]
-        pub signature: Option<Vec<TransactionSignatureElem>>,
-        pub transaction_hash: StarknetTransactionHash,
-        /// None for Invoke and Deploy, Some() for Declare
+        pub nonce: Option<TransactionNonce>,
+        /// Some() for Declare
         #[serde(default)]
         pub sender_address: Option<ContractAddress>,
-        /// None for Invoke and Deploy, Some() for Declare
+        #[serde_as(as = "Option<Vec<TransactionSignatureElemAsDecimalStr>>")]
         #[serde(default)]
-        pub nonce: Option<TransactionNonce>,
+        pub signature: Option<Vec<TransactionSignatureElem>>,
+        pub transaction_hash: StarknetTransactionHash,
         pub r#type: Type,
+        #[serde(default)]
+        pub version: Option<TransactionVersion>,
```

Comment:
Good catch. This change is one more reason why sync will stop with outdated pathfinder versions once 0.9 is deployed to testnet/mainnet.

Reply:
We need to have some tests for integration to pick any quiet changes to the api. These would not be run by the ci but still could be utilized to pick all the changes to the api and utilized before each cairo release.